packages/manager/tools/testcafe/pages/common/managerParent.js

```diff
@@ -0,0 +1,35 @@
+import { Selector, t } from 'testcafe';
+import { getPageUrl } from '../../utils/helpers';
+
+export default class ManagerParentPage {
+  constructor({ currentPageNameInUrl } = {}) {
+    this.currentPageNameInUrl = currentPageNameInUrl;
+    this.navbarUser = Selector('[data-navi-id="user-menu"]');
+    this.logoutLink = Selector('[data-navi-id="logout"]');
+  }
+
+  async confirmCurrentPage() {
+    for (let i = 0; i < 15; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
+      const currentPage = await getPageUrl();
+      if (currentPage.includes(this.currentPageNameInUrl)) {
+        break;
+      }
+    }
```

Comment:
Hmm there's a reason to loop 15 times ? (And not 17 or whatever?)
Also, there is no other solutions to make async call in a for loop ?

Reply:
i add to do it to allow the url some time to change before checking it's value, i used 15  because it was enough for all testcases i trie with some margin

Reply:
Ok seen together